include/boost/math/special_functions/round.hpp

```diff
@@ -14,6 +15,16 @@
 #include <boost/math/policies/error_handling.hpp>
 #include <boost/math/special_functions/math_fwd.hpp>
 #include <boost/math/special_functions/fpclassify.hpp>
+#include <type_traits>
+#include <limits>
+#include <cmath>
+
+#if __cplusplus >= 201703L || _MSVC_LANG >= 201703L
```

Comment:
Minor point, I get nervous about some direct checks like this, should we use BOOST_NO_CXX17_IF_CONSTEXPR instead, or failing that BOOST_CXX_VERSION?

Reply:
BOOST_NO_CXX17_IF_CONSTEXPR works because there is a definition for standalone mode already